### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by jbeard4** [OpenKieler/elkjs#46]
Hi,

I am attempting to build elkjs master, and I am getting the following error:

```jacob@jacob-gpd-pocket:~/workspace/projects/elkjs$ npm run build

> elkjs@0.5.0 build /home/jacob/workspace/projects/elkjs
> npm run gradle && npm run js


> elkjs@0.5.0 gradle /home/jacob/workspace/projects/elkjs
> ./gradlew lib

To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: http://gradle.org/docs/2.5/userguide/gradle_daemon.html.
:copyElkJs UP-TO-DATE
:copyElk UP-TO-DATE
:copyEmul UP-TO-DATE
:generateXtextXtext
ERROR:JsonImportException cannot be resolved. (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 59 column : 13)
ERROR:The method formatError(String) from the type JsonAdapter refers to the missing type Object (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 33 column : 19)
ERROR:The method formatError(String) from the type JsonAdapter refers to the missing type Object (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 42 column : 28)
WARNING:Null-safe call of primitive-valued feature doubleValue, default value 0 will be used (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 106 column : 13)
:generateXtextXtext FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateXtextXtext'.
> Xtext validation failed, see build log for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 28.143 secs
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! elkjs@0.5.0 gradle: `./gradlew lib`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the elkjs@0.5.0 gradle script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jacob/.npm/_logs/2018-08-14T21_12_00_598Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! elkjs@0.5.0 build: `npm run gradle && npm run js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the elkjs@0.5.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jacob/.npm/_logs/2018-08-14T21_12_00_633Z-debug.log
```

elk is also checked out at master. This is after maven has successfully downloaded all package dependencies.

Here is my java information:
```
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.17.04.2-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

Thank you very much for your help with this.

**Comment by uruuru** [comment]
Can you try

./gradlew clean
./gradlew lib

**Comment by jbeard4** [comment]
Thanks for following up. I followed your instructions and the result is the same:

```
jacob@jacob-gpd-pocket:~/workspace/projects/elkjs$ ./gradlew clean
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: http://gradle.org/docs/2.5/userguide/gradle_daemon.html.
:clean

BUILD SUCCESSFUL

Total time: 11.405 secs
jacob@jacob-gpd-pocket:~/workspace/projects/elkjs$ ./gradlew lib
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: http://gradle.org/docs/2.5/userguide/gradle_daemon.html.
:copyElkJs
:copyElk UP-TO-DATE
:copyEmul
:generateXtextXtext
ERROR:JsonImportException cannot be resolved. (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 59 column : 13)
ERROR:The method formatError(String) from the type JsonAdapter refers to the missing type Object (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 33 column : 19)
ERROR:The method formatError(String) from the type JsonAdapter refers to the missing type Object (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 42 column : 28)
WARNING:Null-safe call of primitive-valued feature doubleValue, default value 0 will be used (file:/home/jacob/workspace/projects/elkjs/build/src/elk/org/eclipse/elk/graph/json/JsonAdapter.xtend line : 106 column : 13)
:generateXtextXtext FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateXtextXtext'.
> Xtext validation failed, see build log for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 28.898 secs
```

**Comment by uruuru** [comment]
This line confuses me after a clean (it shouldn't be up to date..):

:copyElk UP-TO-DATE

**Comment by jbeard4** [comment]
I looked at `build.gradle`, and I think I found the issue. The build expects `elk` to exist in the directory above `elkjs`. When I read the README, I read this line to mean that `elk` should exist **inside of** `elkjs`: `a checkout of the ELK repository is required and should be located in the same directory as the checkout of this repository.` I believe moving `elk` to the directory outside of `elkjs` has resolved this issue.

Thank you for your help with this!
